core/tests/performance_framework/perf_domain.py

```diff
@@ -0,0 +1,156 @@
+                raise utils.ValidationError(
+                    'Expected the log entry of the page load stats to include '
+                    'an additional \'entries\' element')
+
+            for entry in self.page_session_stats['log']['entries']:
+                if '_error' in entry['response']:
+                    raise utils.ValidationError(
+                        'Expected a valid server response, found server '
+                        'not reachable.')
+
+            if self.get_request_count() == 0:
+                raise utils.ValidationError(
+                    'Expected the log entry of the page load stats to include '
+                    'a positive number of entries.')
+
+            if self.get_total_page_size_bytes() == 0:
+                raise utils.ValidationError(
+                    'Expected the total size of a page including all its '
+                    'resources to be positive.')
+
+        if self.page_session_timings:
+            for timing_prop in self.TIMING_PROPERTIES:
+                if timing_prop not in self.page_load_timings:
+                    raise utils.ValidationError(
+                        'Expected the timing entry of the page load timings to '
+                        'include %s property' % timing_prop)
+
+                if timing_prop in self.page_session_timings:
+                    if len(str(self.page_session_timings[timing_prop])) < 13:
+                        raise utils.ValidationError(
+                            'Expected %s to be in milliseconds, instead found '
+                            'seconds' % timing_prop)
+
+            if self.get_page_load_time_secs() < 0:
+                raise utils.ValidationError(
+                    'Expected the page load time to be positive.')
+            if self.get_dom_ready_time_secs() < 0:
+                raise utils.ValidationError(
+                    'Expected the dom ready time to be positive.')
+            if self.get_request_time_secs() < 0:
+                raise utils.ValidationError(
+                    'Expected the request time to be positive.')
+
+
+    def get_request_count(self):
+        """Returns the number of requests made prior to the page load
+        completing.
+        """
+        return len(self.page_session_stats['log']['entries'])
+
+    def get_total_page_size_bytes(self):
+        """Returns the total size of a page including all of its resources."""
+        total_size = 0
+
+        for entry in self.page_session_stats['log']['entries']:
+            total_size += int(entry['response']['bodySize'])
+
+        return total_size
+
+    def _get_duration_secs(self, event_end, event_initial):
+        # Timestamps are in milliseconds.
+        initial_timestamp = self.page_load_timings[event_initial]
+
+        end_timestamp = self.page_load_timings[event_end]
+
+        duration_secs = end_timestamp - initial_timestamp
+
+        # Returns in seconds.
```

Comment:
Why? Why not return in milliseconds? If we use milliseconds everywhere instead of seconds, that will keep things consistent and avoid the extra conversion. Dealing directly with milliseconds is quite standard, as well, so there're no additional expectations for the developer reading this.